sound/soc/sof/ipc4-pcm.c

```diff
@@ -215,8 +215,41 @@ static int sof_ipc4_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 	return 0;
 }
 
+static int sof_ipc4_pcm_setup(struct snd_sof_dev *sdev, struct snd_sof_pcm *spcm)
+{
+	struct snd_sof_pipeline_trigger_info *pipeline_info;
+	struct sof_ipc4_fw_data *ipc4_data = sdev->private;
+	int stream;
+
+	for_each_pcm_streams(stream) {
+		pipeline_info = &spcm->stream[stream].pipeline_info;
+
+		/* allocate memory for max number of pipeline IDs */
+		pipeline_info->pipeline_list = kcalloc(ipc4_data->max_num_pipelines,
```

Comment:
nit-pick: is this max - 1?

If this is the pipelines that are connected, then each FE pipeline cannot be part of the list since they are the head?

Reply:
the question is "what does this list represent"?

a) all pipelines reachable from the PCM?
b) all pipelines connected to the first pipeline associated with the PCM.

off-by-one count in the latter case.